From 848ba9f9b376bd2b2e71d7d44643d68b9cbf7031 Mon Sep 17 00:00:00 2001
From: Caesar Wang <wxt@rock-chips.com>
Date: Wed, 13 Dec 2017 17:21:41 +0800
Subject: [PATCH] net: wireless: rockchip_wlan: fixes the warning with gcc-7.2
 build

when used the arm-linux-gnueabihf-gcc-7.2.1, had the below
warning information:

1) drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/siutils.c:607:14:
    warning: self-comparison always evaluates to false
    [-Wtautological-compare]

2) drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_common.c:3583:16:
    warning: comparison between pointer and zero character constant
    [-Wpointer-compare] error, forbidden warning: dhd_common.c:3583

Change-Id: Ida287252b7f20b17b68365b965edbff63ae7cbf9
Signed-off-by: Caesar Wang <wxt@rock-chips.com>
---
 drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_common.c | 2 +-
 drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/siutils.c    | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_common.c b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_common.c
index 19a9226ecd2c..485594e9c308 100644
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_common.c
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_common.c
@@ -3580,7 +3580,7 @@ dhd_pktfilter_offload_set(dhd_pub_t * dhd, char *arg)
 					htod16(WL_PKT_FILTER_MFLAG_NEG);
 				(argv[i])++;
 			}
-			if (argv[i] == '\0') {
+			if (strlen(argv[i]) == 0) {
 				printf("Pattern not provided\n");
 				goto fail;
 			}
diff --git a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/siutils.c b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/siutils.c
index 18fd64a7d5d3..00ae5869a316 100644
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/siutils.c
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/siutils.c
@@ -604,11 +604,14 @@ si_doattach(si_info_t *sii, uint devid, osl_t *osh, volatile void *regs,
 	}
 
 	sih->bustype = bustype;
+
+#ifdef BCMBUSTYPE
 	if (bustype != BUSTYPE(bustype)) {
 		SI_ERROR(("si_doattach: bus type %d does not match configured bus type %d\n",
 			bustype, BUSTYPE(bustype)));
 		return NULL;
 	}
+#endif
 
 	/* bus/core/clk setup for register access */
 	if (!si_buscore_prep(sii, bustype, devid, sdh)) {
-- 
2.35.3

